src/app/register/register.component.ts

```diff
@@ -9,6 +10,20 @@ import { FormControl, FormGroup, Validators } from '@angular/forms';
 export class RegisterComponent {
   registerForm = new FormGroup({
     userName: new FormControl('', Validators.required),
-    password: new FormControl('', Validators.required)
+    password: new FormControl('', Validators.required),
   });
+
+  constructor(private registerService: AuthenticationService) {}
+
+  onSubmit() {
+    if (this.registerForm.valid) {
+      const userName = this.registerForm.value.userName as string;
+      const password = this.registerForm.value.password as string;
+      const role = 2 as number;
```

Comment:
Casting types like this (`as string`, `as number`) can be avoided if we pass the type when creating FormControl above at lines 11-13. For example: `userName: new FormControl<string>('', Validators required)`